docs/apps/hyperqueue.md

```diff
@@ -4,234 +4,280 @@ tags:
+    # Calculate the total memory reservation and convert it from megabytes to bytes.
+    TOTAL_MEM_BYTES=$((SLURM_CPUS_PER_TASK * SLURM_MEM_PER_CPU * 1000000))
+    TOTAL_MEM_OPT="--resource mem=sum($TOTAL_MEM_BYTES)"
+else
+    TOTAL_MEM_OPT=""
+fi
+
+# Start the workers in the background.
+srun --overlap --cpu-bind=none --mpi=none hq worker start \
+    --manager slurm \
+    --on-server-lost finish-running \
+    --cpus="$SLURM_CPUS_PER_TASK" \
+    $TOTAL_MEM_OPT &
+
+# Wait until all workers have started
+hq worker wait "$SLURM_NTASKS"
 ```
 
-`<COMMAND>` then has access to the environment variable `HQ_TASK_ID` which is used
-to enumerate all the tasks.
+**Computing tasks**
 
-When we have submitted everything we want, we need to wait for the jobs to finish.
-This can be done e.g. with:
+Now we can submit tasks with `hq submit` to the server, which executes them on the available workers.
+It is a non-blocking command; thus, we do not need to run it in the background.
+Regarding file I/O, we turn off output by setting `--stdout=none` and `--stderr=none`.
+Otherwise, HyperQueue will create output files for each task, which can create excess I/O on the parallel filesystem when there are many tasks.
+After submitting all the tasks, we wait for them to complete to synchronize the script.
 
 ```bash
+# Submit tasks to workers
+hq submit --stdout=none --stderr=none --cpus=1 --array=1-1000 ./task
+
+# Wait for all the tasks to finish
 hq job wait all
 ```
 
-Once we are done running all of our jobs, we shutdown the workers and server to avoid a false
-error from Slurm when the job ends:
+It is worth reading the sections about [Job and Tasks](https://it4innovations.github.io/hyperqueue/stable/jobs/jobs/) and [Task arrays](https://it4innovations.github.io/hyperqueue/stable/jobs/arrays/) to understand the different ways to run computation with HyperQueue.
+For more complex task dependencies, we can use HyperQueue as the executor for other workflow managers, such as [Snakemake](#using-snakemake-with-hyperqueue) or [Nextflow](#using-nextflow-with-hyperqueue).
+
+**Stopping the workers and the server**
+
+Once we are done running all of our tasks, we shut down the workers and server to avoid a false error from Slurm when the job ends.
 
 ```bash
+# Shut down the workers and server
 hq worker stop all
 hq server stop
 ```
 
-#### sbatch-hq
 
-For simple task farming workflows, where you only want to run many similar (non-MPI parallel,
-independent) commands/programs, you can use the CSC utility tool `sbatch-hq`. Just specify the
-list of commands to run in a file, one command per line. The tool `sbatch-hq` will create and
-launch a batch job that starts running commands from the file using HyperQueue. You can specify
-how many nodes you want to run the commands on and `sbatch-hq` will keep executing the commands
-until all are done, or the batch job time limit is reached.
+### Using local disks with HyperQueue
```

Comment:
I wonder if this should be extended with a more complete batch script example?

Reply:
I can include a more extensive example. I'll add it to the full copy-pasteable scripts.